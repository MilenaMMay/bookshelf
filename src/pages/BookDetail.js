import React from 'react';
import PropTypes from 'prop-types';
import PageTitle from '../components/PageTitle';
import TextContent from '../components/TextContent';
import SecondaryCommand from '../components/SecondaryCommand';

class BookDetail extends React.Component {
  render() {
    return (
      <div>
        <PageTitle>{this.props.title}</PageTitle>
        <TextContent>{this.props.details}</TextContent>
        <SecondaryCommand href={this.props.backUrl}>Back</SecondaryCommand>
      </div>
    );
  }
}

BookDetail.propTypes = {
  title: PropTypes.string,
  details: PropTypes.string,
  backUrl: PropTypes.string
};

export default BookDetail

