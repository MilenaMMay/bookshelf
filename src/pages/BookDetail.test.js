import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import BookDetail from './BookDetail';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BookDetail />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders as specified in the snapshot', () => {
  const component = renderer.create(
    <BookDetail title='Foo' details='Foo bar lorem ipsum dolor foo bar.'/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
