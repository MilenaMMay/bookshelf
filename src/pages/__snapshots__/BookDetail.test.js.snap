// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as specified in the snapshot 1`] = `
<div>
  <h1
    className="sc-bdVaJa cQWkDZ"
  >
    Foo
  </h1>
  <p
    className="sc-bwzfXH kDFfNT"
  >
    Foo bar lorem ipsum dolor foo bar.
  </p>
  <a
    className="sc-htpNat kvAKSG"
  >
    Back
  </a>
</div>
`;
